data/src/main/java/com/linkedin/data/schema/resolver/SchemaDirectory.java

```diff
@@ -0,0 +1,39 @@
+/*
+   Copyright (c) 2020 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.linkedin.data.schema.resolver;
+
+/**
+ * Represents a schema directory relative to the resolver path (directory or Jar file). This is used to customize
+ * schema resolvers to limit which directories are used for resolving schema references.
+ *
+ * @author Karthik Balasubramanian
+ */
+public interface SchemaDirectory
```

Comment:
Suggest to rename to something like "SchemaDirectoryRepresentation" to indicate this could be an interface

Reply:
I prefer to keep this name. It actually captures what this interface represents. 
Naming for interfaces and classes should be similar (i.e, usually nouns). See https://google.github.io/styleguide/javaguide.html#s5.2.2-class-names

Also, IDEs do a good job of indicating when a type is an interface.

Reply:
I just think hard to image I will name an interface as "SchemaDirectory" (which is ideally a class, a noun)